examples/supported_use_cases/leanpipe.cs

```diff
@@ -0,0 +1,5 @@
+using LeanCode.Contracts;
+
+public class Notification : INotification { }
+
+public class Topic : ITopic, IProduceNotification<Notification> { }
```

Comment:
We need more tests, for example:
1. Topic, where the `ITopic` is implemented in the base class,
2. Class that implements `ITopic` but is abstract,
3. Interface that implements `ITopic`,
4. `IProduceNotification` that are implemented by an interface that is then implemented by topic,
5. `IProduceNotification` implemented by a base class.

And probably a little bit more.

Reply:
Definitely, I wanted to validate the approach before writing more of them. Should have stated that in the description.